test/extensions/http/stateful_session/cookie/cookie_test.cc

```diff
@@ -0,0 +1,93 @@
+#include "source/common/network/address_impl.h"
+#include "source/extensions/http/stateful_session/cookie/cookie.h"
+
+#include "test/mocks/upstream/host.h"
+#include "test/test_common/utility.h"
+
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Http {
+namespace StatefulSession {
+namespace Cookie {
+namespace {
+
+TEST(CookieBasedSessionStateFactoryTest, EmptyCookieName) {
+  CookieBasedSessionStateProto config;
+
+  EXPECT_THROW_WITH_MESSAGE(std::make_shared<CookieBasedSessionStateFactory>(config),
+                            EnvoyException,
+                            "Cookie key cannot be empty for cookie based stateful session sticky");
+  config.set_name("override_host");
+
+  EXPECT_NO_THROW(std::make_shared<CookieBasedSessionStateFactory>(config));
+}
+
+TEST(CookieBasedSessionStateFactoryTest, SessionStateTest) {
+  testing::NiceMock<Envoy::Upstream::MockHostDescription> mock_host;
+
+  {
+    CookieBasedSessionStateProto config;
+    config.set_name("override_host");
+    CookieBasedSessionStateFactory factory(config);
+
+    // No valid address in the request headers.
+    Envoy::Http::TestRequestHeaderMapImpl request_headers;
+    auto session_state = factory.create(request_headers);
+    EXPECT_EQ(absl::nullopt, session_state->upstreamAddress());
+
+    auto upstream_host = std::make_shared<Envoy::Network::Address::Ipv4Instance>("1.2.3.4", 80);
+    EXPECT_CALL(mock_host, address()).WillOnce(testing::Return(upstream_host));
+
+    Envoy::Http::TestResponseHeaderMapImpl response_headers;
+    session_state->onUpdate(mock_host, response_headers);
+
+    // No valid address then update it by set-cookie.
+    EXPECT_EQ(response_headers.get_("set-cookie"),
+              Envoy::Http::Utility::makeSetCookieValue("override_host",
+                                                       Envoy::Base64::encode("1.2.3.4:80", 10), "",
```

Comment:
I see we use base64, this will be helpful if we surface this in API docs.

Reply:
Thanks. I will add the comments to the API docs.